File: backend/internal/appdto/knowledge.go
package appdto

import "time"

type CreateExperienceReq struct {
	Type     string `json:"type" validate:"required"`
	Title    string `json:"title" validate:"required"`
	Content  string `json:"content" validate:"required"`
	SourceID string `json:"source_id" validate:"omitempty"`
	Tags     string `json:"tags" validate:"omitempty"` // JSON string
	RoleID   string `json:"role_id" validate:"omitempty"`
}

type UpdateExperienceReq struct {
	ID      string `json:"id" validate:"required"`
	Title   string `json:"title" validate:"omitempty"`
	Content string `json:"content" validate:"omitempty"`
	Tags    string `json:"tags" validate:"omitempty"`
}

type GetExperienceReq struct {
	Page     int    `form:"page" json:"page"`
	PageSize int    `form:"page_size" json:"page_size"`
	Type     string `form:"type" json:"type"`
	Keyword  string `form:"keyword" json:"keyword"`
	Q        string `form:"q" json:"q"`
	RoleID   string `form:"role_id" json:"role_id"`
}

type Experience struct {
	ID         string    `json:"id"`
	Type       string    `json:"type"`
	Title      string    `json:"title"`
	Content    string    `json:"content"`
	SourceID   string    `json:"source_id"`
	Tags       string    `json:"tags"`
	UsageCount int64     `json:"usage_count"`
	CreatedBy  string    `json:"created_by"`
	CreatedAt  time.Time `json:"created_at"`
}

File: frontend/src/apis/knowledge.ts
import { request } from "@/utils";

export interface Experience {
  id: string;
  type: string;
  title: string;
  content: string;
  source_id?: string;
  tags?: string;
  usage_count: number;
  created_at: string;
  created_by: string;
}

export interface CreateExperienceRequest {
  type: string;
  title: string;
  content: string;
  source_id?: string;
  tags?: string;
  role_id?: string;
}

export interface GetExperienceRequest {
  page?: number;
  page_size?: number;
  type?: string;
  keyword?: string;
  role_id?: string;
  q?: string; // Compatible with 'q' parameter
}

export interface UpdateExperienceRequest {
  id: string;
  title?: string;
  content?: string;
  tags?: string;
}

export const getExperiences = (params?: GetExperienceRequest) => request.get<{ list: Experience[]; total: number }>('/experiences/search', { params });
export const createExperience = (data: CreateExperienceRequest) => request.post<{ id: string }>('/experiences', data);
export const getExperience = (id: string) => request.get<Experience>(`/experiences/${id}`);
export const updateExperience = (id: string, data: UpdateExperienceRequest) => request.put(`/experiences/${id}`, data);
export const deleteExperience = (id: string) => request.delete(`/experiences/${id}`);
